### PR TITLE
pwc schedule added 

### PR DESCRIPTION
**Pull request by yashjagtap23**
Added PWC schedule to home page 
(Icon needs to be changed)

Screenshot 2023-07-25 at 5 05 55 PM
Screenshot 2023-07-25 at 5 06 03 PM

### CODE REVIEW

**Comment by aw-0** [comment]
Hey there! Congrats & thank you for your first contribution & PR 🎉 

Both myself & @JosephShepin will look at this and provide feedback on improving the PWC Schedule before we merge it into the website.

At first glance, it'd be great if you could find a different icon for the PWC card (you can find all the options at https://fontawesome.com) that would fit. Additionally, there should be a way to navigate back from the PWC page to the main homepage. You can take a look at the GPA calculator, Jukebox, or Timer for examples of a redirect link back to home.

For @JosephShepin -- do you think this should be a separate page, or should it be a small widget on the homepage/maybe combined into the calendar? Lmk what you think

**Comment by JosephShepin** [comment]
thanks for submitting this PR:
I would really like to see more here than a static page of text. I would like to see making it a card that you can toggle from settings adding more helpful information like "open today until 6pm" or "closed until x/xx" as opposed to just giving the information that is already present on the [stevenson website](https://www.d125.org/student-life/patriot-wellness-center) - if we wanted to do that, we would just add a link to the PWC website in the /links tab. This can be achieved using a lot of the components and code we have written for the schedules system. This is unfortunately going to be a challenging modification to adopt but the point of stevenson.space is to make information that is accessible on stevenson's website presented in a refined and simpler way and I think this is how that can be done. Good luck and let us know if you have any questions!

**Review by JosephShepin — CHANGES_REQUESTED** [review]
check comment

**Comment by aw-0** [comment]
These changes are looking better! Thanks for working with us. I think it'd be great to continue building off of Joey's idea, where you could move the countdown from a separate page to a "widget" on the home screen like the weather/schedule. From there, you can enable/disable viewing in the settings. Feel free to reach out to either of us if you have questions on the implementation of this :)

**Comment by yashjagtap23** [comment]
OK I will work on moving the countdown from a separate page to a "widget" on the home screen now. Thanks for the feedback!

**Comment by yashjagtap23** [comment]
image of countdown as card 
Screenshot 2023-08-02 at 6 39 21 PM

**Comment by aw-0** [comment]
Nice work, thanks for implementing those changes! Few more things I'd personally like to see:

- Having the header match the same font-weight & size as the rest of the widget headers
- The "Time remaining" text could fit in a little better. Maybe make only the countdown timer a bit bigger with the "time remaining until closure" a small subtitle under the countdown?
- What does the UI look like when the PWC is closed? 
- As Joey mentioned, it'd be great to have the ability to toggle the appearance of this card on/off in settings, as not everyone goes to the PWC and/or would want this on

**Review by aw-0 — CHANGES_REQUESTED** [review]
check my earlier comment :)

**Comment by yashjagtap23** [comment]
Fixed styling. However, couldn't figure out how to make a feature so that I could turn the card on or off in settings
Screenshot 2023-08-03 at 2 28 23 PM

**Comment by yashjagtap23** [comment]
green line to match weather 
Screenshot 2023-08-03 at 2 57 56 PM

**Comment by aw-0** [comment]
One more revision I'd like to see super quick - when the PWC is closed, I think it'd be better to just list the time it opens than "opens in 15:12:32", e.x. "opens next at 6 PM". I like the countdown for when it is open (potential psych trick into going there lol), but it's a little confusing when you're trying to see what specific time it's open next.

**Review by aw-0 — APPROVED** [review]
LGTM! Thanks for working with me on those changes

cc @JosephShepin for a final review as well. do we want to have the PWC schedule show on the home screen by default or not?

**Comment by JosephShepin** [comment]
thanks for working on this :can we get colors and font sizes matching?

Screenshot 2023-08-07 at 2 39 06 PM

File: src/components/cards/PwcCard.vue
Comment (JosephShepin) [inline src/components/cards/PwcCard.vue:30]: so we can get rid of this along with the interval. there is already a variable that gets updated every second in our state management system (pinia).

`    ...mapState(useScheduleStore, ['date']),
`

File: src/components/cards/PwcCard.vue
Comment (JosephShepin) [inline src/components/cards/PwcCard.vue:112]: unused function

**Comment by JosephShepin** [comment]
the setting "yes" or "no" doesn't persist between page refreshes

File: src/views/Settings/General.vue
Comment (JosephShepin) [inline src/views/Settings/General.vue:73]: if you want to delete this function, you can by making it inline with the html so like

` :modelValue="selectedPWCState  ? 0 : 1"`

**Review by JosephShepin — CHANGES_REQUESTED** [review]
really nice progress!

File: src/views/Settings/General.vue
Comment (aw-0) [reply src/views/Settings/General.vue:73]: Ended up implementing the state for this cuz it's complicated - copilot just suggested this & I went w it lol

**Comment by yashjagtap23** [comment]
Accidentally removed countdown while open in a previous commit so added that back in and color has been changed to match theme

File: src/components/cards/PwcCard.vue
Comment (yashjagtap23) [reply src/components/cards/PwcCard.vue]: When I deleted this line the countdown stops updating every second and becomes static for some reason :(

**Comment by yashjagtap23** [comment]
Switched from old system to universal date system as per Andrew's suggestion.

**Comment by JosephShepin** [comment]
looks a lot better

- I think we can rename the card "PWC Hours" to just "PWC"
- When you view the PWC on a different day, the card should disappear because there is no system in place to know if the PCW will be open/closed on x days in the future
- we're gonna need some way to control when the pwc is closed like on holidays and breaks. There are also special cases where the PWC closes early.

**Comment by JosephShepin** [comment]
ok i'll approve if we can just hide the pwc card if you are looking at it on a day different than the current one

**Comment by yashjagtap23** [comment]
Screenshot 2023-08-13 at 11 59 40 AM
Screenshot 2023-08-13 at 11 59 45 AM

**Comment by JosephShepin** [comment]
💯

**Comment by yashjagtap23** [comment]
Thanks! Had a lot of fun making this and looking forward to contributing more in the future! 

Ps - like andrew said maybe we should enable the pwc card by default as users likely won't know it exists?

**Pull request merged**
